docs/coffeescript.js

```diff
@@ -0,0 +1,309 @@
+const coffeeSelector = document.getElementById("coffee-Selector");
+const teaSelector = document.getElementById("tea-Selector");
+const dessertSelector = document.getElementById("dessert-Selector");
+const grid = document.getElementById("menu-grid");
+const menuGrid = document.getElementById("menu-grid");
+const menuButton = document.getElementById("coffee-menu");
+const coffee_modal = document.getElementById(".modal");
+const navModal = document.querySelector(".nav-modal");
+let total;
+let lastValue;
+let products;
+let productsLength;
+
+// reading the material.JSON file
+async function fetchData() {
+  const response = await fetch("./material.json");
+  products = await response.json();
+  productsLength = products.length;
+  createMenuGrid(coffeeSelector.value);
+}
```

Comment:
Not sure if it needed to be async, but well, I guess we can simulate it. What would be good to have is error handling - that means - what if the fetch failed? what if the reponse didnt come? if no product, what should happen?

Reply:
I have added error handling in the fetchData function now.